pkg/kubelet/nodestatus/setters_test.go

```diff
@@ -1330,6 +1331,32 @@ func TestMachineInfo(t *testing.T) {
 				},
 			},
 		},
+		{
+			desc: "with swap info",
```

Comment:
I'd add some positive and negative feature gate tests for this.

Reply:
> We should have a test that verifies this behavior if feature gate is turned off. Right now it is assuming that this gate is always true which may not be the case.

Good point!
Addressed in commit [`d9777a2` (#129954)](https://github.com/kubernetes/kubernetes/pull/129954/commits/d9777a2db30bcc063752038082e89d4f327d399d).

> What is the behavior of cadvisor in the case of 0 swap? Is Capacity nil or 0? Maybe we can add a test case for that also.

cadvisor always returns a concrete number that can never be nil.
However, with this PR's implementation we don't populate swap capacity if it's zero: https://github.com/kubernetes/kubernetes/pull/129954/files#diff-ef3b90010597098b0d1f6ae71dcd256d14cffd6998159dcf1308b2c783557d75R359-R362.